src/guest_memory.rs

```diff
@@ -665,6 +667,146 @@ pub trait GuestMemory {
         }
     }
 
+    /// Reads up to `count` bytes from an object and writes them into guest memory at `addr`.
+    ///
+    /// Returns the number of bytes written into guest memory.
+    ///
+    /// # Arguments
+    /// * `addr` - Begin writing at this address.
+    /// * `src` - Copy from `src` into the container.
+    /// * `count` - Copy `count` bytes from `src` into the container.
```

Comment:
To match [std](https://doc.rust-lang.org/std/index.html)::[ptr](https://doc.rust-lang.org/std/ptr/index.html)::[write_volatile](https://doc.rust-lang.org/std/ptr/fn.write_volatile.html#) I would prefer the arguments be named:
```suggestion
    /// # Arguments
    /// * `dest` - Begin writing at this address.
    /// * `src` - Begin reading at this address.
    /// * `count` - Copy this many bytes.
```